examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py

```diff
@@ -0,0 +1,347 @@
+        correct_count = np.sum(label_flatten == preds_flatten)
+        acc = correct_count / len(label_flatten)
+        print(acc)
+
+        return acc
+    
+    # tune
+    if args.tune:
+        from neural_compressor import PostTrainingQuantConfig, quantization
+        from onnxruntime.transformers import optimizer
+        from onnxruntime.transformers.fusion_options import FusionOptions
+
+        train_dataset = load_dataset_from_local(args.data_path, args.model_name_or_path)
+        ort_dataset = ONNXRTDataset(args.model_path, train_dataset)
+
+        from neural_compressor.data import DataLoader as INC_DataLoader
+
+        dataloader = INC_DataLoader(
+            framework="onnxruntime", dataset=ort_dataset, batch_size=args.batch_size
+        )
+
+        model_type = "bert"
+        opt_options = FusionOptions(model_type)
+        opt_options.enable_embed_layer_norm = False
+
+        model_optimizer = optimizer.optimize_model(
+            args.model_path,
+            model_type,
+            num_heads=12,
+            hidden_size=768,
+            optimization_options=opt_options,
+        )
+        model = model_optimizer.model
+
+        # check the optimized model is valid
+        import onnxruntime as ort
+
+        try:
+            ort.InferenceSession(
+                model.SerializeToString(), providers=ort.get_available_providers()
+            )
+        except Exception as e:
+            logger.warning("Optimized model is invalid: {}. ".format(e))
+            logger.warning(
+                "Model optimizer will be skipped. "
+                "Try to upgrade onnxruntime to avoid this error"
+            )
+            model = onnx.load(args.model_path)
+
+        config = PostTrainingQuantConfig(
+            approach="static",
+            quant_level=1,
+            quant_format=args.quant_format,
+            # recipes={"smooth_quant": True, "smooth_quant_args": {"alpha": 0.5}},
+        )
+        q_model = quantization.fit(
+            model,
+            config,
+            eval_func=eval_func,
+            calib_dataloader=dataloader,
+        )
+        q_model.save(args.output_model)
+
+    # benchmark
+    if args.benchmark:
+        import onnx
+        import onnxruntime as ort
+        from neural_compressor.data import DataLoader as INC_DataLoader
+
+        train_dataset = load_dataset_from_local(args.data_path, args.model_name_or_path)
+        ort_dataset = ONNXRTDataset(args.model_path, train_dataset)
+        dataloader = INC_DataLoader(
+            framework="onnxruntime", dataset=ort_dataset, batch_size=args.batch_size
+        )
+        model = onnx.load(args.model_path)
+        if args.mode == "performance":
+            from neural_compressor.benchmark import fit
+            from neural_compressor.config import BenchmarkConfig
+
+            conf = BenchmarkConfig(
+                iteration=100, cores_per_instance=28, num_of_instance=1
```

Comment:
add parameters to make cores_per_instance and num_of_instance are configurable

Reply:
Thx, updated by [0c26397](https://github.com/intel/neural-compressor/pull/1121/commits/0c2639750f33f6b306544dba11175291c4ad691d).